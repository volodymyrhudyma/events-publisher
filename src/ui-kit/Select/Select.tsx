import React, { ReactNode } from 'react';
import Select, { components } from 'react-select';
import { IndicatorProps } from 'react-select/lib/components/indicators';
import { Styles } from 'react-select/lib/styles';
import { ThemeConfig } from 'react-select/lib/theme';
import { ValueType } from 'react-select/lib/types';
import icons from 'src/constants/icons';
import { ISelectOption } from 'src/types';
import Icon from 'src/ui-kit/Icon';

// @ts-ignore
const DropdownIndicatorComponent: ReactNode<DropdownIndicatorComponent> =
  components.DropdownIndicator;

const customStyles: Styles = {
  indicatorSeparator: () => ({}),
  placeholder: base => ({
    ...base,
    color: 'rgb(214, 214, 214)',
  }),
  input: base => ({
    ...base,
    color: '#000',
  }),
  control: styles => ({ ...styles, borderWidth: 2 }),
};

const customTheme: ThemeConfig = theme => ({
  ...theme,
  colors: {
    ...theme.colors,
    neutral20: 'rgb(238, 238, 238)',
  },
});

const errorTheme: ThemeConfig = theme => ({
  ...theme,
  colors: {
    ...theme.colors,
    neutral20: 'rgb(253, 178, 178)',
  },
});

/* tslint:disable-next-line */
const DropdownIndicator = (props: IndicatorProps<any>) => (
  <DropdownIndicatorComponent {...props}>
    <Icon src={icons.arrow} />
  </DropdownIndicatorComponent>
);

interface ICustomSelectProps {
  placeholder: string;
  options: ISelectOption[] | Array<ValueType<ISelectOption>>;
  value: ISelectOption | ValueType<ISelectOption>;
  /* tslint:disable-next-line */
  onChange: (value: any) => void;
  error?: string;
}

const CustomSelect = ({
  placeholder,
  options,
  value,
  onChange,
  error,
}: ICustomSelectProps) => (
  <Select
    value={value}
    onChange={onChange}
    options={options}
    styles={customStyles}
    components={{
      DropdownIndicator,
    }}
    placeholder={placeholder}
    theme={error ? errorTheme : customTheme}
  />
);

CustomSelect.defaultProps = {};

export default CustomSelect;
