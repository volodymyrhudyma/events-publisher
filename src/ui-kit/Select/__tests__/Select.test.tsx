import React from 'react';
import { shallow } from 'enzyme';
import Select from '../Select';

describe('Select component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Select
        placeholder="Placeholder"
        options={[
          { value: 'chocolate', label: 'Chocolate' },
          { value: 'strawberry', label: 'Strawberry' },
          { value: 'vanilla', label: 'Vanilla' },
        ]}
        value={{
          value: 'chocolate',
          label: 'Chocolate',
        }}
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
