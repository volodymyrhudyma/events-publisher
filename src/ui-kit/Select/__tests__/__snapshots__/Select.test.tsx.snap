// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly 1`] = `
<StateManager
  components={
    Object {
      "DropdownIndicator": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Chocolate",
        "value": "chocolate",
      },
      Object {
        "label": "Strawberry",
        "value": "strawberry",
      },
      Object {
        "label": "Vanilla",
        "value": "vanilla",
      },
    ]
  }
  placeholder="Placeholder"
  styles={
    Object {
      "control": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "placeholder": [Function],
    }
  }
  theme={[Function]}
  value={
    Object {
      "label": "Chocolate",
      "value": "chocolate",
    }
  }
/>
`;
